### PR TITLE
base: Screen Stabilization [1/2]

### PR DESCRIPTION
**Pull request by Mustang-ssc**
* New Settings menu integrated into QS itself(long click)
* Thanks to @SuperDroidBond for keeping this alive till Pie

Change-Id: Ibccb8d78d79938f42241ea3690ea66c3aace89ea
Signed-off-by: rituj <ritujbeniwal@gmail.com>
Signed-off-by: Mustang-ssc <mustangssc@protonmail.ch>

### CODE REVIEW

**Comment by gwolf2u** [comment]
please correct PR by force pushing only https://github.com/crdroidandroid/android_frameworks_base/pull/381/commits/ab554f34f04eab4090a79da2acf8e61e62738940

**Comment by Mustang-ssc** [comment]
> please correct PR by force pushing only [ab554f3](https://github.com/crdroidandroid/android_frameworks_base/commit/ab554f34f04eab4090a79da2acf8e61e62738940)

Now should be ok

**Review by gwolf2u — COMMENTED** [review]
@neobuddy89 should we keep current options or implement it in some setting under crDroid Settings?

File: packages/SystemUI/src/com/android/systemui/qs/tiles/ScreenStabilizationTile.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        } else {
+            state.icon = new DrawableIcon(mDisable);
+            state.state = Tile.STATE_INACTIVE;
+        }
+        state.contentDescription = state.label;
+    }
+
+    @Override
+    public int getMetricsCategory() {
+        return MetricsEvent.BLISSIFY;
```
Comment (gwolf2u) [inline packages/SystemUI/src/com/android/systemui/qs/tiles/ScreenStabilizationTile.java:153]: pay attention when cherry-picking

**Review by gwolf2u — CHANGES_REQUESTED** [review]

**Comment by gwolf2u** [comment]
also test before PR

**Comment by gwolf2u** [comment]
and also I see no QS tile 
how to make this work?
are there any dependencies ?

**Comment by SuperDroidBond** [comment]
> 
> 
> and also I see no QS tile
> how to make this work?
> are there any dependencies ?

It's because it's not added in config under Custom Tiles Overlay.

**Comment by gwolf2u** [comment]
> > and also I see no QS tile
> > how to make this work?
> > are there any dependencies ?
> 
> It's because it's not added in config under Custom Tiles Overlay.

Indeed
Missed that one 
Seems to work alright
Ty 👍

**Pull request closed without merging**